crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -122,18 +141,54 @@ impl DWalletMPCManager {
         authority_name: AuthorityName,
         session_id: ObjectID,
     ) -> PeraResult<()> {
+        if self.malicious_actors.contains(&authority_name) {
+            return Ok(());
+        }
         let Some(instance) = self.mpc_instances.get_mut(&session_id) else {
-            error!(
+            warn!(
                 "received a message for instance {:?} which does not exist",
                 session_id
             );
-            // TODO (#261): Punish a validator that sends a message related to a non-existing mpc instance
+            self.malicious_actors.insert(authority_name);
             return Ok(());
         };
-        instance.handle_message(DWalletMPCMessage {
+        let handle_message_response = instance.handle_message(DWalletMPCMessage {
             message: message.to_vec(),
             authority: authority_name,
-        })
+        });
+        if let Err(PeraError::DWalletMPCMaliciousParties(malicious_parties)) =
+            handle_message_response
+        {
+            self.store_malicious_parties_by_indices(malicious_parties)?;
+            return Ok(());
+        };
+        handle_message_response
+    }
+
+    /// Convert the indices of the malicious parties to their addresses and store them
+    /// in the malicious actors set
+    /// New messages from these parties will be ignored
+    fn store_malicious_parties_by_indices(
```

Comment:
1. IIRC, there is no function `store_malicious_parties` (i.e., without using indices). Consider renaming to `store_malicious_parties`.

2. "storing" is perhaps not the right phrase here. Consider using `flag_parties_as_malicious` or `report_parties_as_malicious`